Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -419,255 +421,145 @@ namespace AtomSampleViewer
-
-                // Skip if we only have warnings only and we don't want to show warnings
-                skipReport = skipReport || (scriptPassed && scriptHasWarnings && !showWarnings);
-
-                if (skipReport)
-                {
-                    continue;
-                }
-
-                ImGuiTreeNodeFlags scriptNodeFlag = scriptPassed ? FlagDefaultClosed : FlagDefaultOpen;
-
-                AZStd::string header = AZStd::string::format("%s %s",
-                    scriptPassed ? "PASSED" : "FAILED",
-                    scriptReport.m_scriptAssetPath.c_str()
-                );
-
-                HighlightTextFailedOrWarning(!scriptPassed, scriptHasWarnings);
-
-                if (ImGui::TreeNodeEx(&scriptReport, scriptNodeFlag, "%s", header.c_str()))
+                for (ScriptReport& scriptReport : m_scriptReports)
                 {
-                    ResetTextHighlight();
-
-                    // Number of Asserts
-                    HighlightTextIf(scriptReport.m_assertCount > 0, m_highlightSettings.m_highlightFailed);
-                    if (showAll || scriptReport.m_assertCount > 0)
-                    {
-                        ImGui::Text("Asserts:  %u %s", scriptReport.m_assertCount, SeeConsole(scriptReport.m_assertCount, "Trace::Assert").c_str());
-                    }
+                    const bool scriptPassed = scriptReport.m_assertCount == 0 && scriptReport.m_generalErrorCount == 0 && scriptReport.m_screenshotErrorCount == 0;
+                    const bool scriptHasWarnings = scriptReport.m_generalWarningCount > 0 || scriptReport.m_screenshotWarningCount > 0;
 
-                    // Number of Errors
-                    HighlightTextIf(scriptReport.m_generalErrorCount > 0, m_highlightSettings.m_highlightFailed);
-                    if (showAll || scriptReport.m_generalErrorCount > 0)
-                    {
-                        ImGui::Text("Errors:   %u %s", scriptReport.m_generalErrorCount, SeeConsole(scriptReport.m_generalErrorCount, "Trace::Error").c_str());
-                    }
-
-                    // Number of Warnings
-                    HighlightTextIf(scriptReport.m_generalWarningCount > 0, m_highlightSettings.m_highlightWarning);
-                    if (showAll || (showWarnings && scriptReport.m_generalWarningCount > 0))
-                    {
-                        ImGui::Text("Warnings: %u %s", scriptReport.m_generalWarningCount, SeeConsole(scriptReport.m_generalWarningCount, "Trace::Warning").c_str());
-                    }
+                    // Skip if tests passed without warnings and we don't want to show successes
+                    bool skipReport = (scriptPassed && !scriptHasWarnings && !m_showAll);
 
-                    ResetTextHighlight();
+                    // Skip if we only have warnings only and we don't want to show warnings
+                    skipReport = skipReport || (scriptPassed && scriptHasWarnings && !m_showWarnings);
```

Comment:
This 2 lines can be combined into:
scriptPassed && ((!scriptHasWarnings && !m_showAll) || ( scriptHasWarnings && !m_showWarnings));

But if I understand it correctly: m_showAll includes m_showWarnings, then it can be simplified into:
!(!scriptPassed || m_showAll || (scriptHasWarnings && m_showWarnings));

Reply:
Refactored to one line